instrumentation/instaredigo/instaredigo.go

```diff
@@ -0,0 +1,315 @@
+// (c) Copyright IBM Corp. 2022
+
+//go:build go1.16
+// +build go1.16
+
+package instaredigo
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/gomodule/redigo/redis"
+	instana "github.com/instana/go-sensor"
+	ot "github.com/opentracing/opentracing-go"
+	otlog "github.com/opentracing/opentracing-go/log"
+)
+
+type instaRedigoConn struct {
+	redis.Conn
+	sensor   *instana.Sensor
+	address  string
+	prevSpan *prevSpan
+}
+
+type prevSpan struct {
+	span      ot.Span
+	batchCmds []string
+}
+
+//DoContext sends a command to server and returns the received reply along with 
+//the instrumentation.
+func (c *instaRedigoConn) DoContext(ctx context.Context, commandName string,
+	args ...interface{}) (reply interface{}, err error) {
+	if cwc, ok := c.Conn.(redis.ConnWithContext); ok {
+		if c.prevSpan != nil {
```

Comment:
What do think about extracting this `if/else` statement? It is almost the same in other "Do" methods.